### DISCUSSION TITLE
New Tutorial

### DISCUSSION THREAD

**Discussion by RhetTbull** [RhetTbull/osxphotos#435] [Show and tell]
I've written a new Tutorial for the README that I hope will make osxphotos easier for new users.  I welcome any feedback, suggestions, corrections that anyone has!

https://github.com/RhetTbull/osxphotos#tutorial

**Comment by digital821** [comment]
Hey Rhet,

super new to a lot of this. Can understand basics but honestly a little lost. Installed osxphotos on a computer that I am setting up for my wife. Want to pull the old photos out of the library but apple photos won't export anything. Not sure why but I've exhausted that route. 

Basically osxphotos can pull up meta data of the library but export won't work. 

I put in osxphotos export and terminal tells me 'DEST' is missing. I have the directory path but either I'm not sure how to put it in correctly or there's an issue. I assume it can find the library since I can pull the meta data, ex. geo tags and albums works fine. 

If you have any insight it would be greatly appreciated

**Reply by RhetTbull** [reply]
Hi @digital821 Please copy the entire command line you're using with osxphotos in Terminal here and I'll see what I can figure out.

**Reply by RhetTbull** [reply]
>  Installed osxphotos on a computer that I am setting up for my wife.

Also, is the Photos library linked to iCloud?  You said you're setting up a new computer so it's possible  that if this is an iCloud photos library, the photos aren't yet downloaded from iCloud.

**Reply by digital821** [reply]
It's possible about the iCloud but if I can see them in the library can't I export them? I'm taking this computer over from a relative and its not possible to get the iCloud info at this point.

**Reply by digital821** [reply]
I've blanked out my wife's username for privacy other than that it's all the same


J***s-MacBook-Pro:~ j*********$ osxphotos export
Usage: osxphotos export [OPTIONS] [PHOTOS_LIBRARY]... DEST
Try 'osxphotos export -h' for help.

Error: Missing argument 'DEST'.
J***s-MacBook-Pro:~ j*********$ osxphotos
Usage: osxphotos [OPTIONS] COMMAND [ARGS]...

Options:
  --db <Photos database path>  Specify Photos database path. Path to Photos
                               library/database can be specified using either
                               --db or directly as PHOTOS_LIBRARY positional
                               argument. If neither --db or PHOTOS_LIBRARY
                               provided, will attempt to find the library to
                               use in the following order: 1. last opened
                               library, 2. system library, 3.
                               ~/Pictures/Photos Library.photoslibrary
  --json                       Print output in JSON format.
  -v, --version                Show the version and exit.
  -h, --help                   Show this message and exit.

Commands:
  about     Print information about osxphotos including license.
  albums    Print out albums found in the Photos library.
  dump      Print list of all photos & associated info from the Photos...
  export    Export photos from the Photos database.
  help      Print help; for help on commands: help <command>.
  info      Print out descriptive info of the Photos library database.
  keywords  Print out keywords found in the Photos library.
  labels    Print out image classification labels found in the Photos...
  list      Print list of Photos libraries found on the system.
  persons   Print out persons (faces) found in the Photos library.
  places    Print out places found in the Photos library.
  query     Query the Photos database using 1 or more search options; if...
  repl      Run interactive osxphotos shell
  tutorial  Display osxphotos tutorial.
J***s-MacBook-Pro:~ j*********$ osxphotos export
Usage: osxphotos export [OPTIONS] [PHOTOS_LIBRARY]... DEST
Try 'osxphotos export -h' for help.

Error: Missing argument 'DEST'.
J***s-MacBook-Pro:~ j*********$

**Reply by RhetTbull** [reply]
> It's possible about the iCloud but if I can see them in the library can't I export them? 

Not necessarily.  You'd be able to see a preview of the image  (which Photos always keeps) but there are a couple of conditions under which the original images could be missing and thus you couldn't export:

1. Look in Photos | Preferences | General and ensure "Copy Items to the Photos Library" is checked. If not, Photos leaves the photos in their original location when importing to Photos.  If the original location was on an external disk for example, those photos might be missing on this computer. 
2. Also in Photos | Preferences | iCloud see if "iCloud Photos" is checked and if so, is the line that says "Download Originals to this Mac" checked? If not and the alternate "Optimize Mac Storage" is checked, the originals will be in iCloud and only a prevew on the Mac until you try to view or export it at which time Photos will download from iCloud.

Regarding the error with "Missing argument DEST", that stands for "Destination Path" and you need to tell osxphotos where to export the photos (the destination path).  For example, if you wanted the photos exported to "/Users/your_wifes_name/Desktop/photos" you could do the following at the Terminal prompt:

`mkdir ~/Desktop/photos`
`osxphotos export ~/Desktop/photos`

Now, this will export the photos if they're present on the Mac but it's not likely to be exactly what you want since it will dump all the photos into the single directory `~/Desktop/photos`.  Here's another command line that might get you closer:

`osxphotos export ~/Desktop/photos --export-by-date --download-missing`

This exports the photos to ~/Desktop/export
`--export-by-date` creates folders for the date each photo was taken, e.g. `2020/09/20`
`--download-missing` attempts to download any photos missing off the local Mac from iCloud (only works if the Photos library is actually an iCloud library)

If this isn't an iCloud library or you don't have the iCloud password, you can still download the preview image for any missing images using this command:

`osxphotos export ~/Desktop/photos --export-by-date --preview-if-missing`

`--preview-if-missing` instructs osxphotos to export the preview image if the original is missing off the local Mac

Have you read the osxphotos tutorial?  If not, I suggest you do so:

`osxphotos tutorial`

**Reply by digital821** [reply]
Regarding your first two points - I was able to confirm 1. but not 2.


> 1. Look in Photos | Preferences | General and ensure "Copy Items to the Photos Library" is checked. If not, Photos leaves the photos in their original location when importing to Photos.  If the original location was on an external disk for example, those photos might be missing on this computer.
> 2. Also in Photos | Preferences | iCloud see if "iCloud Photos" is checked and if so, is the line that says "Download Originals to this Mac" checked? If not and the alternate "Optimize Mac Storage" is checked, the originals will be in iCloud and only a prevew on the Mac until you try to view or export it at which time Photos will download from iCloud.
> 

I've read the tutorial and I think my issue is I just don't know enough about command lines and how to write them out to get the actions I want. Anyway, adding the DEST path worked! Kind of. Heres what happened:

J***s-MacBook-Pro:~ j****$ osxphotos export ~/desktop/photos
Using last opened Photos library: /Volumes/STORAGE/Editor/Photos Library.photoslibrary
2021-09-20 07:48:17,085 - WARNING - photosdb.py - 100 - WARNING: This module has only been tested with macOS versions [10.12, 10.13, 10.14, 10.15, 10.16, 11.0, 11.1, 11.2, 11.3, 11.4]: you have Darwin, OS version: 11.5
Exporting 1066 photos to /Users/j*****/desktop/photos...
  [####################################]  100%          
Processed: 1066 photos, exported: 1, missing: 1065, error: 0
Elapsed time: 2.302 seconds

A folder was created on my desktop but only one file was exported. It was a video file, random.


I then input the preview if missing command and here are the results:

J***s-MacBook-Pro:~ j****t$ osxphotos export ~/desktop/photos --export-by-date --preview-if-missing
Using last opened Photos library: /Volumes/STORAGE/Editor/Photos Library.photoslibrary
2021-09-20 07:51:15,211 - WARNING - photosdb.py - 100 - WARNING: This module has only been tested with macOS versions [10.12, 10.13, 10.14, 10.15, 10.16, 11.0, 11.1, 11.2, 11.3, 11.4]: you have Darwin, OS version: 11.5
Exporting 1066 photos to /Users/j****/desktop/photos...
  [###########-------------------------]   31%  00:00:13Error exporting photo (350399FB-4D8E-4D75-915D-5939662492D2: Theoneguitar2.jpg) as Theoneguitar2.jpg: Error Domain=NSCocoaErrorDomain Code=516 "“350399FB-4D8E-4D75-915D-5939662492D2_1_105_c.jpeg” couldn’t be copied to “17” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/3/350399FB-4D8E-4D75-915D-5939662492D2_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2018/12/17/Theoneguitar2_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/3/350399FB-4D8E-4D75-915D-5939662492D2_1_105_c.jpeg, NSUnderlyingError=0x7ffbac958fd0 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [#############-----------------------]   37%  00:00:12Error exporting photo (18D53EA8-49FD-40ED-B7A0-D7A66A0E4961: Theoneguitar2.jpg) as Theoneguitar2.jpg: Error Domain=NSCocoaErrorDomain Code=516 "“18D53EA8-49FD-40ED-B7A0-D7A66A0E4961_1_105_c.jpeg” couldn’t be copied to “17” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/1/18D53EA8-49FD-40ED-B7A0-D7A66A0E4961_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2018/12/17/Theoneguitar2_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/1/18D53EA8-49FD-40ED-B7A0-D7A66A0E4961_1_105_c.jpeg, NSUnderlyingError=0x7ffbac8cd6b0 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
Error exporting photo (26E331AA-A697-443B-9F7A-BD50E25E85B8: IMG_4946.JPG) as IMG_4946.JPG: Error Domain=NSCocoaErrorDomain Code=516 "“26E331AA-A697-443B-9F7A-BD50E25E85B8_1_105_c.jpeg” couldn’t be copied to “20” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/2/26E331AA-A697-443B-9F7A-BD50E25E85B8_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2018/06/20/IMG_4946_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/2/26E331AA-A697-443B-9F7A-BD50E25E85B8_1_105_c.jpeg, NSUnderlyingError=0x7ffbac96e1b0 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [###############---------------------]   44%  00:00:10Error exporting photo (955EA008-F326-4480-9DD5-303183F4E0BC: IMG_1805c.jpg) as IMG_1805c.jpg: Error Domain=NSCocoaErrorDomain Code=516 "“955EA008-F326-4480-9DD5-303183F4E0BC_1_105_c.jpeg” couldn’t be copied to “12” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/9/955EA008-F326-4480-9DD5-303183F4E0BC_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2018/04/12/IMG_1805c_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/9/955EA008-F326-4480-9DD5-303183F4E0BC_1_105_c.jpeg, NSUnderlyingError=0x7ffbac948320 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [##################------------------]   50%  00:00:09Error exporting photo (2656304A-1CC9-4A4A-A30E-A6E0C3F9A61E: IMG_8827.JPG) as IMG_8827.JPG: Error Domain=NSCocoaErrorDomain Code=516 "“2656304A-1CC9-4A4A-A30E-A6E0C3F9A61E_1_105_c.jpeg” couldn’t be copied to “06” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/2/2656304A-1CC9-4A4A-A30E-A6E0C3F9A61E_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2019/04/06/IMG_8827_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/2/2656304A-1CC9-4A4A-A30E-A6E0C3F9A61E_1_105_c.jpeg, NSUnderlyingError=0x7ffbac97e850 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [##################------------------]   52%  00:00:08Error exporting photo (8D8FDFA2-6F86-4031-8CFD-91FF23D8ECA5: IMG_6109.JPG) as IMG_6109.JPG: Error Domain=NSCocoaErrorDomain Code=516 "“8D8FDFA2-6F86-4031-8CFD-91FF23D8ECA5_1_105_c.jpeg” couldn’t be copied to “16” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/8/8D8FDFA2-6F86-4031-8CFD-91FF23D8ECA5_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2018/09/16/IMG_6109_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/8/8D8FDFA2-6F86-4031-8CFD-91FF23D8ECA5_1_105_c.jpeg, NSUnderlyingError=0x7ffbac947980 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [###################-----------------]   52%  00:00:08Error exporting photo (CFBE677A-1FA9-4BDA-8260-2FB5472141E2: IMG_6110.JPG) as IMG_6110.JPG: Error Domain=NSCocoaErrorDomain Code=516 "“CFBE677A-1FA9-4BDA-8260-2FB5472141E2_1_105_c.jpeg” couldn’t be copied to “16” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/C/CFBE677A-1FA9-4BDA-8260-2FB5472141E2_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2018/09/16/IMG_6110_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/C/CFBE677A-1FA9-4BDA-8260-2FB5472141E2_1_105_c.jpeg, NSUnderlyingError=0x7ffbac8bda80 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [#####################---------------]   58%  00:00:07Error exporting photo (371D9633-40FA-4DF3-8DC4-08A82E971A3A: IMG_3912.jpg) as IMG_3912.jpg: Error Domain=NSCocoaErrorDomain Code=516 "“371D9633-40FA-4DF3-8DC4-08A82E971A3A_1_105_c.jpeg” couldn’t be copied to “20” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/3/371D9633-40FA-4DF3-8DC4-08A82E971A3A_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2019/10/20/IMG_3912_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/3/371D9633-40FA-4DF3-8DC4-08A82E971A3A_1_105_c.jpeg, NSUnderlyingError=0x7ffbac94ea60 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [#####################---------------]   59%  00:00:07Error exporting photo (8AC1641B-2AFF-4ECB-98F2-B83805D9DFAC: IMG_5479.JPG) as IMG_5479.JPG: Error Domain=NSCocoaErrorDomain Code=516 "“8AC1641B-2AFF-4ECB-98F2-B83805D9DFAC_1_105_c.jpeg” couldn’t be copied to “29” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/8/8AC1641B-2AFF-4ECB-98F2-B83805D9DFAC_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2018/07/29/IMG_5479_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/8/8AC1641B-2AFF-4ECB-98F2-B83805D9DFAC_1_105_c.jpeg, NSUnderlyingError=0x7ffbac98b470 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [#########################-----------]   71%  00:00:05Error exporting photo (8EB9EFAB-BFE3-4FEE-9C2D-5E3279114F49: IMG_3231.HEIC) as IMG_3231.HEIC: Error Domain=NSCocoaErrorDomain Code=516 "“8EB9EFAB-BFE3-4FEE-9C2D-5E3279114F49_1_105_c.jpeg” couldn’t be copied to “24” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/8/8EB9EFAB-BFE3-4FEE-9C2D-5E3279114F49_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2019/07/24/IMG_3231_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/8/8EB9EFAB-BFE3-4FEE-9C2D-5E3279114F49_1_105_c.jpeg, NSUnderlyingError=0x7ffbac99a3a0 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
Error exporting photo (A628AFE6-129C-46AF-8CE9-CFCBFED145EE: IMG_3301.HEIC) as IMG_3301.HEIC: Error Domain=NSCocoaErrorDomain Code=516 "“A628AFE6-129C-46AF-8CE9-CFCBFED145EE_1_105_c.jpeg” couldn’t be copied to “26” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/A/A628AFE6-129C-46AF-8CE9-CFCBFED145EE_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2019/07/26/IMG_3301_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/A/A628AFE6-129C-46AF-8CE9-CFCBFED145EE_1_105_c.jpeg, NSUnderlyingError=0x7ffbac8addc0 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [##########################----------]   73%  00:00:04Error exporting photo (E46B5F53-4F6A-491D-9EBE-A7FA567D1531: IMG_3303.HEIC) as IMG_3303.HEIC: Error Domain=NSCocoaErrorDomain Code=516 "“E46B5F53-4F6A-491D-9EBE-A7FA567D1531_1_105_c.jpeg” couldn’t be copied to “26” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/E/E46B5F53-4F6A-491D-9EBE-A7FA567D1531_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2019/07/26/IMG_3303_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/E/E46B5F53-4F6A-491D-9EBE-A7FA567D1531_1_105_c.jpeg, NSUnderlyingError=0x7ffbacc3dd10 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [############################--------]   79%  00:00:03Error exporting photo (EE0FDD3F-959E-4A72-BC2C-3E9D29CAC04E: IMG_4975.JPG) as IMG_4975.JPG: Error Domain=NSCocoaErrorDomain Code=516 "“EE0FDD3F-959E-4A72-BC2C-3E9D29CAC04E_1_105_c.jpeg” couldn’t be copied to “20” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/E/EE0FDD3F-959E-4A72-BC2C-3E9D29CAC04E_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2018/06/20/IMG_4975_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/E/EE0FDD3F-959E-4A72-BC2C-3E9D29CAC04E_1_105_c.jpeg, NSUnderlyingError=0x7ffbacc64ff0 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [##############################------]   83%  00:00:02Error exporting photo (4A074191-DA79-4248-9D2E-6737F1DF349B: dmd6.jpg) as dmd6.jpg: Error Domain=NSCocoaErrorDomain Code=516 "“4A074191-DA79-4248-9D2E-6737F1DF349B_1_105_c.jpeg” couldn’t be copied to “08” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/4/4A074191-DA79-4248-9D2E-6737F1DF349B_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2018/05/08/dmd6_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/4/4A074191-DA79-4248-9D2E-6737F1DF349B_1_105_c.jpeg, NSUnderlyingError=0x7ffbac8c2b40 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [###############################-----]   88%  00:00:01Error exporting photo (16162F7F-DED9-45CF-8833-4D6EBA52B854: DSC04306.jpg) as DSC04306.jpg: Error Domain=NSCocoaErrorDomain Code=516 "“16162F7F-DED9-45CF-8833-4D6EBA52B854_1_105_c.jpeg” couldn’t be copied to “17” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/1/16162F7F-DED9-45CF-8833-4D6EBA52B854_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2018/12/17/DSC04306_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/1/16162F7F-DED9-45CF-8833-4D6EBA52B854_1_105_c.jpeg, NSUnderlyingError=0x7ffba9f70f00 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [################################----]   89%  00:00:01Error exporting photo (B75420FF-EE5C-4090-BFC8-4CA8AB37D9EE: IMG_4947.JPG) as IMG_4947.JPG: Error Domain=NSCocoaErrorDomain Code=516 "“B75420FF-EE5C-4090-BFC8-4CA8AB37D9EE_1_105_c.jpeg” couldn’t be copied to “20” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/B/B75420FF-EE5C-4090-BFC8-4CA8AB37D9EE_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2018/06/20/IMG_4947_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/B/B75420FF-EE5C-4090-BFC8-4CA8AB37D9EE_1_105_c.jpeg, NSUnderlyingError=0x7ffbac9d23b0 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [################################----]   90%  00:00:01Error exporting photo (BADDDCF7-91DD-46F3-A1E2-7D091454C320: FullSizeRender.jpg) as FullSizeRender.jpg: Error Domain=NSCocoaErrorDomain Code=516 "“BADDDCF7-91DD-46F3-A1E2-7D091454C320_1_105_c.jpeg” couldn’t be copied to “09” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/B/BADDDCF7-91DD-46F3-A1E2-7D091454C320_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2019/12/09/FullSizeRender_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/B/BADDDCF7-91DD-46F3-A1E2-7D091454C320_1_105_c.jpeg, NSUnderlyingError=0x7ffbac8eac60 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [################################----]   91%  00:00:01Error exporting photo (93A66454-3D01-4C46-9444-174BA9C3CBDE: IMG_3302.HEIC) as IMG_3302.HEIC: Error Domain=NSCocoaErrorDomain Code=516 "“93A66454-3D01-4C46-9444-174BA9C3CBDE_1_105_c.jpeg” couldn’t be copied to “26” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/9/93A66454-3D01-4C46-9444-174BA9C3CBDE_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/2019/07/26/IMG_3302_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/9/93A66454-3D01-4C46-9444-174BA9C3CBDE_1_105_c.jpeg, NSUnderlyingError=0x7ffbac9bd930 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
  [####################################]  100%          
Processed: 1066 photos, exported: 1038, missing: 0, error: 18
Elapsed time: 17.598 seconds
J****s-MacBook-Pro:~ j****$


After this, the 1038 files exported onto my desktop in a photos folder, they're preview size which I'm okay with but none of the videos exported, obviously. Im fine with what I've got so I thank you immensely but if there is a way to get the videos and other misc off the library I'd love to try.

Cheers for everything

**Reply by RhetTbull** [reply]
I believe the copy error you're getting (due to images with the same name being dumped in the same folder) were fixed recently (#515).  Not sure how you installed osxphotos but try updating and running the export again:

Assuming you used pipx:
`pipx upgrade osxphotos`

The try running the export again, using the --update flag:

`osxphotos export ~/desktop/photos --update`

If you still get those errors, it's a bug, so let me know and I'll take a look.

As for the videos, nothing I can do there as Photos doesn't usually store a preview of the video file, only a low-res screen grab that it shows in Photos until it can start streaming the video from the cloud. If you can log in to iCloud, you can use the `--download-missing` option to force download of those videos, otherwise you're out of luck.

**Comment by digital821** [comment]
Did the upgrade, thank you.

tried download missing and got this error for every photo. Two videos did come through though. Very weird.

This was repeated for over 1000 photos. I ended up downloading the previews, which worked. I guess the iCloud is probably the main culprit

Error exporting photo (C82DF29C-E0E5-42E2-93E8-43A5872F1DF5: IMG_3310.HEIC) as IMG_3310.HEIC: Error Domain=NSCocoaErrorDomain Code=516 "“C82DF29C-E0E5-42E2-93E8-43A5872F1DF5_1_105_c.jpeg” couldn’t be copied to “photos” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/C/C82DF29C-E0E5-42E2-93E8-43A5872F1DF5_1_105_c.jpeg, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/j****/desktop/photos/IMG_3310_preview.jpeg, NSFilePath=/Volumes/STORAGE/Editor/Photos Library.photoslibrary/resources/derivatives/C/C82DF29C-E0E5-42E2-93E8-43A5872F1DF5_1_105_c.jpeg, NSUnderlyingError=0x7fbda9b3e600 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}

**Reply by RhetTbull** [reply]
I was able to replicate this behavior. It's a bug - you shouldn't get those copy errors.

However, if you used --update you shouldn't have seen those errors.  (I could only replicate this if I didn't use --update).  --update doesn't copy files that are already exported, only files not yet exported.

**Reply by RhetTbull** [reply]
I've opened a bug (#516) for the copy error (you got these because the preview file already existed but that shouldn't have happened as osxphotos should have incremented the file name when the destination file already exists).  e.g. IMG_3310_preview.jpeg should have been incremented as IMG_3310 (1)_preview.jpeg.  I believe --update would have avoided this (it skips files already exported) but either way, this is still a bug.  

For your use case, despite the copy errors, I believe you likely have extracted all you can from the Photos library at this point since you don't have the iCloud credentials.

**Reply by RhetTbull** [reply]
I've fixed this bug with preview images in [v0.42.84](https://github.com/RhetTbull/osxphotos/releases/tag/v0.42.84)
